lib/livebook/settings.ex

```diff
@@ -117,4 +120,160 @@ defmodule Livebook.Settings do
   def set_update_check_enabled(enabled) do
     storage().insert(:settings, "global", update_check_enabled: enabled)
   end
+
+  @doc """
+  Gets a list of environment variables from storage.
+  """
+  @spec fetch_env_vars() :: list(EnvironmentVariable.t())
+  def fetch_env_vars do
+    for fields <- storage().all(:environment_variables) do
+      struct!(EnvironmentVariable, fields)
+    end
+  end
+
+  @doc """
+  Gets one environment variable from storage.
+
+  Raises `RuntimeError` if the environment variable does not exist.
+  """
+  @spec fetch_env_var!(String.t()) :: EnvironmentVariable.t()
+  def fetch_env_var!(id) do
+    case storage().fetch(:environment_variables, id) do
+      :error -> raise RuntimeError, "the environment variable #{id} does not exists in storage"
+      {:ok, fields} -> struct!(EnvironmentVariable, fields)
+    end
+  end
+
+  @doc """
+  Checks if environment variable already exists.
+  """
+  @spec env_var_exists?(String.t()) :: boolean()
+  def env_var_exists?(key) do
+    Enum.any?(fetch_env_vars(), &(&1.key == key))
```

Comment:
```suggestion
    storage().fetch(:environment_variables, id) != :error
```

Reply:
In this case, we are searching by `key` and not `id`, because the ID is autogenerated by the `EnvironmentVariable.changeset/2`.

So, my idea was, listing all and searching by `key`. Could this be a callback inside `Livebook.Storage`?

Reply:
I think we should make the key the ID, since the key is unique anyway?